lib/json/string_parser.nit

```diff
@@ -0,0 +1,288 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Simple ad-hoc implementation of a JSON parser for String inputs
+module string_parser
+
+import parser_base
+import static
+
+redef class Char
+	# Is `self` a valid number start ?
+	fun is_json_num_start: Bool do
+		if self == '-' then return true
+		if self.is_numeric then return true
+		return false
+	end
+
+	# Is `self` a valid JSON separator ?
+	fun is_json_separator: Bool do
+		if self == ':' then return true
+		if self == ',' then return true
+		if self == '{' then return true
+		if self == '}' then return true
+		if self == '[' then return true
+		if self == ']' then return true
+		if self == '"' then return true
+		if self.is_whitespace then return true
+		return false
+	end
+end
+
+# A simple ad-hoc JSON parser
+class JSONStringParser
+	super StringProcessor
+
+	# Parses an JSON Entity
+	#
+	# ~~~nit
+	#	var p = new JSONStringParser("\{\"numbers\": [1,23,3], \"string\": \"string\"\}")
+	#	assert p.parse_entity isa JsonObject
+	# ~~~
```

Comment:
*a JSON entity...

While reopening this doc, it would be worth it to use a triple quoted string instead of all the escaping.